src/ts/generic/featurecollection.ts

```diff
@@ -184,6 +185,30 @@ export function buildHeader(header: HeaderMeta, crsCode = 0): Uint8Array {
     return builder.asUint8Array() as Uint8Array;
 }
 
+
+export async function readMetadata(
+    url: string,
+    nocache = false,
+    headers: HeadersInit = {}    
+): Promise<HeaderMeta> {
+    
+    //defined in ../http-reader.ts
+    const assumedHeaderLength = 2024;
+    const httpClient = new HttpRangeClient(url, nocache, headers);
+
+    const bytes = new Uint8Array(await httpClient.getRange(0, assumedHeaderLength, 'read metadata'));
+
+    if (!bytes.subarray(0, 3).every((v, i) => magicbytes[i] === v)) throw new Error('Not a FlatGeobuf file');
+
+    const bb = new flatbuffers.ByteBuffer(bytes);
+    const headerLength = bb.readUint32(magicbytes.length);
+    bb.setPosition(magicbytes.length + SIZE_PREFIX_LEN);
```

Comment:
The problem is here, because of changed/fixed logic in fromByteBuffer, the position move should not include SIZE_PREFIX_LEN.